otj-server-core/src/main/java/com/opentable/server/PortSelector.java

```diff
@@ -92,80 +101,62 @@ public PortSelector(Environment environment) {
         this.environment = environment;
     }
 
-    private PortSelection get(
-                                      String springPropertyName,
-                                      int ordinal,
-                                      String portName) {
-        // First the named port - hence on k8s this gets precedence
-        PortSelection portSelection = get(portName, PortSource.FROM_PORT_NAMED);
-        if (!portSelection.hasValue()) {
-            // Then the "official" spring property
-            portSelection =  get(springPropertyName, PortSource.FROM_SPRING_PROPERTY);
-            if (!portSelection.hasValue()) {
-                // Finally the PORTn using a best guess here.
-                portSelection = get("PORT" + ordinal, PortSource.FROM_PORT_ORDINAL);
+    private PortSelection get(String springPropertyName, String namedPort) {
+        PortSelection portSelection = get(springPropertyName, springPropertyName, PortSource.FROM_SPRING_PROPERTY);
```

Comment:
We need to check is property supplied first

Reply:
But that goes against the NAMED PORT first, the core port of this PR